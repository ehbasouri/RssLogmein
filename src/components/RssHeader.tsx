import React from 'react';
import {Text, View} from 'react-native';
import {componentStyles} from './componentStyles';

interface Props {
  title: string | undefined;
  onBackPress: () => void | undefined;
  showBack?: boolean | undefined;
}

export function RssHeader({title, showBack, onBackPress}: Props) {
  return (
    <View style={componentStyles.headerContainer}>
      <Text onPress={onBackPress} style={componentStyles.back}>
        {showBack && 'BACK'}
      </Text>
      <Text style={componentStyles.title}>{title}</Text>
      <View style={componentStyles.back} />
    </View>
  );
}
