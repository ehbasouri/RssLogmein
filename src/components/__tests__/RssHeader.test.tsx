import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {RssHeader} from '../RssHeader';

describe('Rss Header tests', () => {
  it('Header test', () => {
    let props: any = {
      onBackPress: jest.fn(),
      title: 'title',
      showBack: true,
    };
    const {toJSON, getByText} = render(<RssHeader {...props} />);

    const item = getByText('BACK');
    fireEvent.press(item);
    expect(props.onBackPress).toHaveBeenCalled();
    const title = getByText('title');
    expect(title).toBeTruthy();
    expect(toJSON).toMatchSnapshot();
  });
});
