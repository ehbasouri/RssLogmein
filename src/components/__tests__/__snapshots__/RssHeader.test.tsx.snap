// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rss Header tests Header test 1`] = `[Function]`;
